content/post/22-04-24-progress-report-33.md

```diff
@@ -0,0 +1,283 @@
+
+{{< newsection >}}
+## Content
+### April Stress Test
+On April 9th we hosted a stress test with as many players as possible to see how well the server could handle it, peaking at 250 players! Normally April events would go in the April progress report, but we're including it here while it's hopefully still fresh in all of the participants' minds.
+
+Space Station 14 was able to run with 200 players without any server slowdown. The stress test peaked at ~250 players and we predict that it could have surpassed ~350 without slowdown. However, although the server can handle it, we noticed some networking issues that can cause rubberbanding with this many players. The probable cause has been identified and efforts to fix it are underway.
+
+For a highlight reel of the chaos caused by 250 players, you can watch the YouTube video [here](https://www.youtube.com/watch?v=__5K6pWj1Ls).
+
+<details>
+    <summary>See here for a more technical breakdown of performance</summary>
+    
+There are two parts to how the game processes things: A unit of time called a "tick" during which the game simulates everything, and the "ticks per second (TPS)" which is the rate at which ticks occur. For example, 30 TPS means that one second is divided into 30 ticks of equal length and the game will attempt to update 30 times per second. More ticks per second means less time for each tick, and if the game isn't finished running the code by the time the tick ends then the game will lag. On the flip side, a higher TPS means the game will feel more responsive. Choosing the right TPS is a balancing act, and the upper limit mainly depends on how performant the game is.
+    
+Space Station 14 was able to run at our usual 30 TPS with over 200 players largely without issue. After lowering it to 20 TPS, which is what most SS13 servers use, the stress test peaked at ~250 players and continued to perform fine (aka everything was able to run during the allotted ticks).
+</details>
+
+tl;dr SS14 can definitely handle over 250 players and can probably handle over 350 players before the server starts to lag. Unless you do something wild like give everyone mechs with rocket launchers. Not too shabby, in my opinion.
+
+{{< newsection >}}
+### Explosion Refactor
+*Contributed by ElectroSR*
+
+Explosions have been completely refactored to be more visually and technically impressive. There's also a new UI for admins to create explosions and it comes with a very fancy preview of what will be annihilated.
+
+{{< video-audio "/video/pr_33/explosions.mp4" >}}
+
+{{< newsection >}}
+### Modular Grenades
+*Contributed by ElectroSR and Fortune117*
+
+Fortune117 and ElectroSR added every Chemist's favorite tool - the modular grenade, which allows anyone with a screwdriver and some basic parts to create a grenade filled with anything from deadly acid smoke to healing chemicals.
+
+TODO: Video?
```

Comment:
Probably easiest to just take the video from the #progress-reports channel

https://discord.com/channels/310555209753690112/310560938438819861/955233312656461864